### ISSUE TITLE
[Question] Is there an equivalent to Terragrunts `dependency` block?

### ISSUE THREAD

**Issue by donovanrost** [terramate-io/terramate#991]
Terragrunt offers an opinionated way to pass values from dependency to dependent using the [dependency block](https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#dependency). 

This feature makes it easy for sibling level modules to pass values from one to the other. 
One example of this is passing database information from a database module into a web service which requires that configuration data.

What is the `terramate` equivalent feature or technique to accomplish this?

**Comment by soerenmartius** [comment]
Thanks for the excellent question @donovanrost. One of the reasons why we invented Terramate is that we discourage using outputs which is the de facto standard used in Terragrunt. To mention some of the reasons:

- Outputs are hard-coded and user-defined contracts. If you change the output scheme in the providing module, all consumers must adopt it.
- Outputs are values that are persisted in Terraforms state file. This can often lead to issues where outputs become stale because of, e.g., drifts.

We managed dozens of mid to large-scale platforms using Terragrunts' approach and often ran into frustrating situations, hence the decision for Terramate to follow a different approach.

In Terramate, we recommend the following approaches:

- If you need to share data among a range of stacks, we recommend you use [Globals](https://terramate.io/docs/cli/data-sharing/#globals). Globals allow you to establish data that can be reused across multiple stacks using hierarchical merge semantics. This ensures consistent and easy data sharing within your project.
- If you need to pass data from "sibling level modules" from one to another, instead of using outputs, we recommend you use [Data Sources](https://developer.hashicorp.com/terraform/language/data-sources). Data Sources trigger actual API calls instead of looking up values in Terraforms' state.

You might also be interested in https://github.com/terramate-io/terramate/discussions/571

PS: I am about to publish a series of blog posts covering the recommended approaches.

**Comment by soerenmartius** [comment]
Hey @donovanrost,

Please let me know if that answered your question and if there's anything else I can help you with! Happy weekend!

**Comment by donovanrost** [comment]
Yes that did answer my question, thank you. I do think it would be cool to express a dependency between to modules and have terramate do the Data Source calls under the hood. I don't think that would be inline with the product goals

**Comment by soerenmartius** [comment]
> Yes that did answer my question, thank you. I do think it would be cool to express a dependency between to modules and have terramate do the Data Source calls under the hood. I don't think that would be inline with the product goals

Hey @donovanrost, thanks for your feedback. We're working on an improved version of the orchestration and are also thinking about ways to improve the UX using data sources. What's important to remember, though is that the orchestration functionality isn't tight to Terraform exclusively. You can basically orchestrate all commands (e.g. `infracost` or `kubectl`). 

We will discuss this during our next product planning session and get back to you with feedback!
